src/minimization.cpp

```diff
@@ -0,0 +1,289 @@
+//===----------------------------------------------------------------------===/ /
+//
+// Part of the Utopia EDA Project, under the Apache License v2.0
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2023 ISP RAS (http://www.ispras.ru)
+//
+//===----------------------------------------------------------------------===//
+
+#include "layout.h"
+
+#include <vector>
+
+typedef std::pair<TreeNode *, TreeNode *> Edge;
+
+namespace {
+  struct AdditionalNetFeatures {
+    std::vector<std::vector<TreeNode::Id>> nodesByLayer;
+    std::vector<std::vector<TreeNode::Id>> tempNodesByLayer;
+    std::vector<std::vector<Edge>> netEdges;
+    int intersections = -1;
+    std::vector<int> accTree;
+
+    void minimizeIntersections(Net &net);
+    void setEdgesToOptimalCondition(Net &net);
+    int crossCounting(AdditionalNetFeatures &features);
```

Comment:
Redundant parameter, instead of `features.crossCounting(features)` use `features.crossCounting()`